File: branches/V1.0/proj/CoreLibrary/src/com/jasonzqshen/familyaccounting/core/exception/MasterDataIdentityNotDefined.java
package com.jasonzqshen.familyaccounting.core.exception;

import com.jasonzqshen.familyaccounting.core.masterdata.MasterDataIdentity;
import com.jasonzqshen.familyaccounting.core.masterdata.MasterDataType;

public class MasterDataIdentityNotDefined extends Exception {

	/**
	 * 
	 */
	private static final long serialVersionUID = -1012865754518579084L;
	private final MasterDataIdentity _identity;
	private final MasterDataType _type;

	public MasterDataIdentityNotDefined(MasterDataIdentity id,
			MasterDataType type) {
		super(String.format("Bank Key %s is not defined %s", id.toString(),
				type.toString()));

		_identity = id;
		_type = type;
	}

	/**
	 * get master data identity
	 * 
	 * @return
	 */
	public MasterDataIdentity getIdentity() {
		return _identity;
	}

	/**
	 * get master data type
	 * 
	 * @return
	 */
	public MasterDataType getMasterDataType() {
		return _type;
	}
}

File: branches/V1.0/proj/WinStore_FamilyAccounting/WinStore_FamilyAccountingCore/Exceptions/MasterDataIdentityNotDefined.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WinStore_FamilyAccountingCore.MasterData;

namespace WinStore_FamilyAccountingCore.Exceptions
{
    public class MasterDataIdentityNotDefined : Exception
    {
        public MasterDataIdentity MdId { get; protected set; }
        public MasterDataType MdType { get; protected set; }
        public MasterDataIdentityNotDefined(MasterDataIdentity id,
            MasterDataType type)
            : base(String.Format("Bank Key %s is not defined %s", id.ToString(),
                type.ToString()))
        {
            MdId = id;
            MdType = type;
        }
    }
}
